scanner/walk_dir_tree.go

```diff
@@ -214,7 +214,9 @@ func loadDir(ctx context.Context, job *scanJob, dirPath string, ignorePatterns [
 				folder.numPlaylists++
 			case model.IsImageFile(entry.Name()):
 				folder.imageFiles[entry.Name()] = entry
-				if fileInfo.ModTime().After(folder.imagesUpdatedAt) {
+				if folder.imagesUpdatedAt.IsZero() {
+					folder.imagesUpdatedAt = folder.modTime
+				} else if fileInfo.ModTime().After(folder.imagesUpdatedAt) {
 					folder.imagesUpdatedAt = fileInfo.ModTime()
 				}
```

Comment:
Shouldn't this be something like: 
```suggestion
				if folder.imagesUpdatedAt.IsZero() || fileInfo.ModTime().After(folder.imagesUpdatedAt) {
					folder.imagesUpdatedAt = timeNewest(folder.imagesUpdatedAt, fileInfo.ModTime(), folder.modTime)
				}
```

```
func timeNewest(times ...time.Time) time.Time {
	newest := time.Time{}
	for _, t := range times {
		if t.After(newest) {
			newest = t
		}
	}
	return newest
}
```

Reply:
I suppose it makes things clearer, but it doesn't change the underlying concern of changing an image to one that has an older mod time

Reply:
Upon further review, I mistakenly thought that `imagesUpdatedAt` came from the DB (which it does not). It does mean that if the folder updates (but not image), then image cache will always be refreshed.
I don't know how to best approach that.

Reply:
> It does mean that if the folder updates (but not image), then image cache will always be refreshed.
> I don't know how to best approach that.

I think that's fine.

Reply:
> Upon further review, I mistakenly thought that imagesUpdatedAt came from the DB (which it does not). It does mean that if the folder updates (but not image), then image cache will always be refreshed.
I don't know how to best approach that.

What about the `timeNewest` idea i posted above? It would get the newest of any image found in the folder or the modified time of the folder itself. Have you tried it?

Reply:
I mean, that would work, but would result in refreshing the image cache whenever the folder is updated.

Reply:
> I mean, that would work, but would result in refreshing the image cache whenever the folder is updated.

That's a small price to pay. How often we expect a folder to change in a music library? Once we use hashes, this could be fixed anyway.

Reply:
As long as moving/removing items in the folder updates the mod time, that should be good _enough_ for now.

Reply:
I opted for a much easier change (just default `imagesUpdatedAt` to folder updated at). Requires that the underlying filesystem file addition/removal changes the folder mod time.